### ISSUE TITLE
Can't add app_role value as null

### ISSUE THREAD

**Issue by guri-s** [terraform-providers/terraform-provider-azuread#130]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azuread_application`

### Terraform Configuration Files

```hcl
resource "azuread_application" "xxxx" {
  name                       = "xxxxx"
  homepage                   = "https://sso.online.xxxxxxx.com"
  available_to_other_tenants = false
  oauth2_allow_implicit_flow = false

  reply_urls = [
    "https://sso.online.xxxxxxx.com,
  ]

  app_role = [
    {
      "allowed_member_types" = ["User"]
      "description"          = "msiam_access"
      "display_name"         = "msiam_access"
      "is_enabled"           = true
      "value"                = ""
    },
    {
      allowed_member_types = ["User"]
      "description"        = "User"
      "display_name"       = "User"
      "is_enabled"         = true
      "value"              = ""
    },
  ]
}

```

### Debug Output

We should be able to set a null value for the `app_role`.
If I look at the app manifest in Azure the value is 'null'. But it is not allowed to be set in tf.
Below is from azure manifest of the app
```
"appRoles": [
		{
			"allowedMemberTypes": [
				"User"
			],
			"description": "User",
			"displayName": "User",
			"id": "xxxx-xx-xx-x--xxxx-xxx",
			"isEnabled": true,
			"lang": null,
			"origin": "Application",
			"value": null
		},
		{
			"allowedMemberTypes": [
				"User"
			],
			"description": "msiam_access",
			"displayName": "msiam_access",
			"id": "xxxxxx-xxxx-xxxx",
			"isEnabled": true,
			"lang": null,
			"origin": "Application",
			"value": null
		}
	],
```

### Expected Behavior

We should be able to set the app_role as null

### Actual Behavior

```Error: azuread_application.xxxxx: "app_role.0.value" must not be empty```

### Steps to Reproduce

1. `terraform plan`

**Comment by enorlando** [comment]
Hi, Is anyone able to shed some light on this issue?

**Comment by matsest** [comment]
Looks like the error comes from the validation function NoEmptyStrings

https://github.com/terraform-providers/terraform-provider-azuread/blob/83d4ab741630a3bd1236b5c4863703ea982aad30/azuread/helpers/validate/strings.go#L9-L21

which is called in the definition of value

https://github.com/terraform-providers/terraform-provider-azuread/blob/7e185411a66e3c0d6556df86def092b16431ab66/azuread/resource_application.go#L139-L143

I guess we either need to have a different validation which allows for null in app_role.value.

**Comment by matsest** [comment]
In addition, since you have
https://github.com/terraform-providers/terraform-provider-azuread/blob/4fdb3ae9dcf6a730a4fddc88b1f5cb0415ca1dc0/azuread/resource_application.go#L103-L105

you can't set `id` manually - it will automatically delete the approle and create a new with a computed id.

**Comment by guri-s** [comment]
I see that v.0.6.0 was released last week. Can we expect this to be fixed in the next release?

**Comment by enorlando** [comment]
hi @katbyte, Do we have an eta on a fix for this issue? Thanks

**Comment by enorlando** [comment]
Hi @katbyte , do we have an eta on when this bug will be fixed?
Thank you!

**Comment by guri-s** [comment]
Hi, can someone please advise if this one is in the pipeline to be fixed. We just have a lot of resources affected by this and would be good to know if there is a fix coming soon.

**Comment by katbyte** [comment]
@guri-s & @enorlando this seems like an easy enough fix to get in for 0.7.0
